src/Components/Components/test/NavigationManagerTest.cs

```diff
@@ -1,10 +1,17 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+using System.Diagnostics;
+using Microsoft.AspNetCore.Components.Routing;
+
 namespace Microsoft.AspNetCore.Components;
 
 public class NavigationManagerTest
```

Comment:
Tests look great and thorough; I would suggest adding a couple of additional tests where the handler throws an exception both synchronously and asynchronously to capture that behavior, since it is something that will likely happen on users apps.

Reply:
Just added some tests that do this